src/test/java/medic_appointment_app/appointment/AppointmentControllerTest.java

```diff
@@ -0,0 +1,166 @@
+package medic_appointment_app.appointment;
+
+import com.clinic.medic_appointment_app.PostgresContainerFunctionalTestBase;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.*;
+
+import java.time.LocalDateTime;
+
+class AppointmentControllerTest extends PostgresContainerFunctionalTestBase {
+
+    @Test
+    void create_HappyPath() {
+        AppointmentDTO dto = new AppointmentDTO();
+        dto.setUserId(1L);
+        dto.setScheduleId(1L);
+        dto.setReason("Checkup");
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<AppointmentDTO> request = new HttpEntity<>(dto, headers);
+
+        ResponseEntity<AppointmentDO> response = restTemplate
+                .withBasicAuth("patient@example.com", "patientpass")
+                .postForEntity(getBaseHost() + "/api/appointments", request, AppointmentDO.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(response.getBody()).isNotNull();
+        assertThat(response.getBody().getReason()).isEqualTo("Checkup");
+    }
+
+    @Test
+    void create_Erro() {
+        AppointmentDTO dto = new AppointmentDTO();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<AppointmentDTO> request = new HttpEntity<>(dto, headers);
+
+        ResponseEntity<String> response = restTemplate
+                .withBasicAuth("patient@example.com", "patientpass")
+                .postForEntity(getBaseHost() + "/api/appointments", request, String.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    void list_HappyPath() {
```

Comment:
please try to use the usual java method naming convention for tests too + add the test prefix/suffix, applicable for all others
```suggestion
    void testListHappyPath() {
```